pulumi/infra/ec2_cluster.py

```diff
@@ -0,0 +1,109 @@
+import itertools
+from typing import Dict, List, Optional, Sequence, Union
+
+import pulumi_aws as aws
+from infra.network import Network
+from typing_extensions import Literal
+
+import pulumi
+from pulumi import Output
+
+# Note that a Quorum of 1 will lose data across updates, not suitable
+# for databases but fine for stateless services
+QuorumSize = Union[Literal[1], Literal[3], Literal[5]]
+
+
+class Ec2Cluster(pulumi.ComponentResource):
+    def __init__(
+        self,
+        name: str,
+        vpc: Network,
+        quorum_size: QuorumSize,
+        quorums: int,  # Number of quorums to actually build
```

Comment:
These names are confusing. A quorum refers to the size of the smallest majority in a cluster. With a cluster of 5, the quorum is 3.

I think `quorum_size` is actually `cluster_size`; I guess `quorums` is actually how many clusters you want to build, with one cluster per private network, relying on the fact that our private networks are in different availability zones?

Do these clusters form a "super-cluster"?

Reply:
Yeah. I renamed the verbiage to "num_subclusters", "subcluster_size"

Reply:
would it be less confusing to just have one number and leave the responsibility of incrementing it properly to the operator?